iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/compaction/evaluator/IcebergCompactionEvaluator.java

```diff
@@ -52,37 +52,33 @@ private IcebergCompactionEvaluator() {
 
   private static final Logger LOG = LoggerFactory.getLogger(IcebergCompactionEvaluator.class);
 
-  public static boolean isEligibleForCompaction(Table icebergTable, String partitionPath,
-      CompactionType compactionType, HiveConf conf) {
+  public static boolean isEligibleForCompaction(Table icebergTable, CompactionInfo ci, HiveConf conf) {
 
     if (icebergTable.currentSnapshot() == null) {
       LOG.info("Table {}{} doesn't require compaction because it is empty", icebergTable,
-          partitionPath == null ? "" : " partition " + partitionPath);
+          ci.partName == null ? "" : " partition " + ci.partName);
       return false;
     }
 
-    CommonPartitionEvaluator partitionEvaluator = createCommonPartitionEvaluator(icebergTable, partitionPath, conf);
+    CommonPartitionEvaluator partitionEvaluator = createCommonPartitionEvaluator(icebergTable, ci, conf);
 
     if (partitionEvaluator == null) {
       return false;
     }
 
-    switch (compactionType) {
+    switch (ci.type) {
       case MINOR:
-        return partitionEvaluator.isMinorNecessary();
+        return partitionEvaluator.isMinorNecessary() || partitionEvaluator.isMajorNecessary();
```

Comment:
can we add some description what `MINOR` and `MAJOR` mean in context of Iceberg?

Reply:
I changed this part mapping Amoro minor to Hive minor and Amoro Major to Hive major.